### PR TITLE
M3-4388 Responsive styling for Entity Header (All variants)

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

This PR adjusts responsive styling for both variants of the entity header (landing and details). Please test various breakpoints thoroughly, and make use of chrome's device emulator and/or Xcode to get an idea of what things will look like for particular devices. I did test 'in-between' breakpoints and made fixes where I could, but in order to get everything working for both variants there are sure to be some less-than-optimal styling decisions I made that weren't included in the designs.

## Type of Change
- Non breaking change ('update')

## Note to Reviewers

Notes will be throughout to explain some of the changes!

### CODE REVIEW

File: packages/manager/src/MainContent_CMR.tsx [hunk 1]
```diff
@@ -59,8 +59,8 @@ const useStyles = makeStyles((theme: Theme) => ({
     transition: theme.transitions.create('opacity'),
     [theme.breakpoints.down('sm')]: {
       paddingTop: theme.spacing(2),
-      paddingLeft: theme.spacing(2),
-      paddingRight: theme.spacing(2)
+      paddingLeft: 0,
```
Comment (WilkinsKa1) [inline packages/manager/src/MainContent_CMR.tsx:62]: Needed to remove this here so tablet views would be accurate. This is technically part of the work for the main content width.

File: packages/manager/src/components/EntityHeader/EntityHeader.tsx [hunk 2]
```diff
@@ -37,6 +49,10 @@ const useStyles = makeStyles((theme: Theme) => ({
       fontSize: '.875rem',
       letterSpacing: '.5px'
     }
+  },
+  bodyDetailVariant: {
```
Comment (WilkinsKa1) [inline packages/manager/src/components/EntityHeader/EntityHeader.tsx:53]: This is for the details variant as the name implies- still also inheriting from contentOuter (the OG class) that landing has as well.

File: packages/manager/src/components/TagCell/TagCell.tsx [hunk 3]
```diff
@@ -13,6 +13,12 @@ const useStyles = makeStyles((theme: Theme) => ({
   root: {
     position: 'relative'
   },
+  rootDetails: {
```
Comment (WilkinsKa1) [inline packages/manager/src/components/TagCell/TagCell.tsx:16]: More conditional styling based on context within the app- we're doing quite a bit of this for CMR.

File: packages/manager/src/components/EntityHeader/EntityHeader.tsx [hunk 4]
```diff
@@ -53,20 +69,23 @@ export const EntityHeader: React.FC<HeaderProps> = props => {
...
+        })}
+      >
+        <HeaderBreadCrumb
+          iconType={iconType}
+          title={title}
+          parentLink={parentLink}
+          parentText={parentText}
+        />
+
+        <Hidden smDown>
```
Comment (Jskobos) [inline packages/manager/src/components/EntityHeader/EntityHeader.tsx:88]: This is causing /domains to crash, bc you're not allowed to have a render function that doesn't render; to render nothing, you have to explicitly return `null`. Has to be `body ? <Grid ..../> : null`

**Comment by WilkinsKa1** [comment]
Thanks for the feedback @Jskobos ! I addressed the issues you called out- I decided to wrap the clear filters button on smaller screens, lmk what you think.

**Comment by WilkinsKa1** [comment]
Thanks for the reviews, @johnwcallahan @tiffwong ! I’ve addressed the following:

• Domains landing
• wrapping for in-between breakpoints
• IP Table header
• Dark mode divider
• Clear filters alignment

Good question in re breakpoints! The lowest I look at is iPhone 6/7/8 plus (this goes for current CM as well).

**Comment by johnwcallahan** [comment]
Thanks @WilkinsKa1, adjustments look good. One thing I didn't notice before: in grid view (landing variant) components aren't collapsing the way they are on the detail pages:
Screen Shot 2020-07-29 at 8 47 45 AM

**Comment by WilkinsKa1** [comment]
@johnwcallahan thanks for checking that- I didn't even realize that wasn't quite the details variant. Made adjustments to account for that as well!

File: packages/manager/src/components/EntityHeader/EntityHeader.tsx [hunk 5]
```diff
@@ -3,32 +3,57 @@ import * as classnames from 'classnames';
...
 import HeaderBreadCrumb, { BreadCrumbProps } from './HeaderBreadCrumb';
+import Hidden from '../core/Hidden';
 
 export interface HeaderProps extends BreadCrumbProps {
   actions?: JSX.Element;
   body?: JSX.Element;
   title: string | JSX.Element;
   bodyClassName?: string;
   isLanding?: boolean;
+  isSecondary?: boolean;
```
Comment (WilkinsKa1) [inline packages/manager/src/components/EntityHeader/EntityHeader.tsx:14]: Note 2 'minor' variants here- one is for 'secondary' landings, such as Ip Addresses table header, and the isDetailsLanding is for card view, which is actually a variation of the true details view as it does not have a breadcrumb and has additional elements we needed to account for.

**Pull request merged**